docs/overview.rst

```diff
@@ -1,11 +1,410 @@
+...            SequentialExampleScheme(examples=8),
+...            ShuffledExampleScheme(examples=8)]
+>>> for scheme in schemes:
+...     print(list(scheme.get_request_iterator()))
+[[0, 1, 2, 3], [4, 5, 6, 7]]
+[[7, 2, 1, 6], [0, 4, 3, 5]]
+[0, 1, 2, 3, 4, 5, 6, 7]
+[7, 2, 1, 6, 0, 4, 3, 5]
+
+We can therefore use an iteration scheme to visit a dataset in some order.
+
+>>> state = dataset.open()
+>>> scheme = ShuffledScheme(examples=dataset.num_examples, batch_size=4)
+>>> for request in scheme.get_request_iterator():
+...     data = dataset.get_data(state=state, request=request)
+...     print(data[0].shape, data[1].shape)
+(4, 2, 2) (4, 1)
+(4, 2, 2) (4, 1)
+>>> dataset.close(state)
+
+Data streams: automating the iteration process
+----------------------------------------------
+
+.. admonition:: In summary
+    :class: tip
+
+    * :class:`AbstractDataStream` subclasses are responsible for coordinating
+      a dataset and an iteration scheme to iterate through the data.
+    * :class:`AbstractDataStream`'s :meth:`get_epoch_iterator` method returns
+      an iterator that returns examples or batches of examples.
+    * :class:`DataStream` is the most common data stream. Its constructor
+      accepts a ``dataset`` argument and an ``iteration_scheme`` argument.
+
+Iteration schemes offer a more convenient way to visit the dataset than
+accessing the data by hand, but we can do better: the act of getting a fresh
+state from the dataset, getting a request iterator from the iteration scheme,
+using both to access the data and closing the state is repetitive. To automate
+this, we have *data streams*, which are subclasses of
+:class:`AbstractDataStream`.
+
+The most common :class:`AbstractDataStream` subclass is :class:`DataStream`. It
+is instantiated with a dataset and an iteration scheme, and returns an epoch
+iterator through its :meth:`get_epoch_iterator` method, which iterates over the
+dataset in the order defined by the iteration scheme.
+
+>>> from fuel.streams import DataStream
+>>> data_stream = DataStream(dataset=dataset, iteration_scheme=scheme)
+>>> for data in data_stream.get_epoch_iterator():
+...     print(data[0].shape, data[1].shape)
+(4, 2, 2) (4, 1)
+(4, 2, 2) (4, 1)
+
+
+Transformers: apply some transformation on the fly
+--------------------------------------------------
+
+.. admonition:: In summary
+    :class: tip
+
+    * :class:`Transformer` subclasses are responsible for taking data stream(s)
+      as input and producing a data stream as output, which applies some
+      transformation to the input stream(s).
+    * Transformers can be chained together to form complex data processing
+      pipelines.
+
+Some data streams take data streams as input. We call them *transformers*, and
+they enable us to build complex data preprocessing pipelines.
+
+Transformers are :class:`Transformer` subclasses. Most of the the transformers
+you'll encounter are located in the ``fuel.transformers`` module. Here are some
```

Comment:
You didn't mention in which module to find `DataStream`s, `Dataset`s, etc, .. before, I wouldn't do it here either. Also, mentioning that most of them are there makes me curious on where to find the rest of them. The link to the class documentation and the example should be enough.